File: LeetCode/11. Container With Most Water.cpp
// Author : Adarsh Gaur

class Solution {
public:
    int maxArea(vector<int>& a) {
        int i=0, j = a.size()-1, area = 0;
        while(i < j){
            if(a[i] < a[j]){
                area = max(area, (j - i)*a[i++]);
            }else{
                area = max(area, (j- i)*a[j--]);
            }
        }
        return area;
    }
};

File: LeetCode/11. Container With Most Water.java
class Solution {
    public int maxArea(int[] height) {
        int maxWater = 0;
        int leftIndex = 0, rightIndex = height.length -1;
        while(leftIndex < rightIndex){
            if(height[leftIndex] < height[rightIndex]){
                maxWater = Math.max(maxWater, height[leftIndex++] * (rightIndex - leftIndex +1));
            }else{
                maxWater = Math.max(maxWater, height[rightIndex--] * (rightIndex - leftIndex +1));
            }
        }
        return maxWater;
    }
}

